demos/python_demos/text_detection_demo_opencv/text_detection_demo_opencv.py

```diff
@@ -0,0 +1,266 @@
+
+    def _mask_to_bboxes(self, min_area=300, min_height=10):
+        image_h, image_w = self.image_shape
+        self.bboxes = []
+        max_bbox_idx = self.mask.max()
+        mask_tmp = cv2.resize(self.mask, (image_w, image_h), interpolation=cv2.INTER_NEAREST)
+
+        for bbox_idx in range(1, max_bbox_idx + 1):
+            bbox_mask = mask_tmp == bbox_idx
+            cnts, _ = cv2.findContours(bbox_mask.astype(np.uint8), cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)
+            if len(cnts) == 0:
+                continue
+            cnt = cnts[0]
+            rect, w, h = self._min_area_rect(cnt)
+            if min(w, h) < min_height:
+                continue
+            if w * h < min_area:
+                continue
+            self.bboxes.append(self._order_points(rect))
+
+    def _min_area_rect(self, cnt):
+        rect = cv2.minAreaRect(cnt)
+        w, h = rect[1]
+        box = cv2.boxPoints(rect)
+        box = np.int0(box)
+        return box, w, h
+
+    def _order_points(self, rect):
+        """ (x, y)
+            Order: TL, TR, BR, BL
+        """
+        tmp = np.zeros_like(rect)
+        sums = rect.sum(axis=1)
+        tmp[0] = rect[np.argmin(sums)]
+        tmp[2] = rect[np.argmax(sums)]
+        diff = np.diff(rect, axis=1)
+        tmp[1] = rect[np.argmin(diff)]
+        tmp[3] = rect[np.argmax(diff)]
+        return tmp
+
+    def decode(self):
+        "Decode loaded data to ``self.bboxes``"
+        for point in self.points:
+            y, x = point
+            neighbours = self._get_neighbours(x, y)
+            for n_idx, nx, ny in neighbours:
+                link_value = self.link_mask[y, x, n_idx]
+                pixel_cls = self.pixel_mask[ny, nx]
+                if link_value and pixel_cls:
+                    self._join(point, (ny, nx))
+
+        self._get_all()
+        self._mask_to_bboxes()
+
+    def plot_result_cvgui(self, image):
+        "Plot and show decoded results via OpenCV's GUI"
+        for box in self.bboxes:
+            cv2.drawContours(image, [box], 0, (0, 0, 255), 2)
+        if args.test is None:
+            cv2.imshow('Detected text', image)
+            if cv2.waitKey():
+                cv2.destroyAllWindows()
+
+
+def main():
+    if args.model_path.endswith('.xml'):
+        td = cv2.dnn.readNet(args.model_path, args.model_path[:-3] + 'bin')
+    else:
+        print("Model's XML file expected")
+        return 1
+
+    img = cv2.imread(args.image_path)
+    if img is None:
+        print("Failed to load image")
+        return 1
+
+    out_layer_names = td.getUnconnectedOutLayersNames()
+    expected_out_layer_names = (['model/link_logits_/add', 'model/segm_logits/add'],
+                                ['pixel_cls/add_2', 'pixel_link/add_2'])
+    if out_layer_names not in expected_out_layer_names:
+        print("Net has unexpected output layer names, please check model files")
+        print(f"Expected: '{expected_out_layer_names}', returned: '{out_layer_names}'")
```

Comment:
f-strings are only supported since Python 3.6, and we have to support Python 3.5. Please replace them with explicit `format` calls.

Reply:
done